runtime/onert/core/src/util/ShapeInference.cc

```diff
@@ -961,17 +961,16 @@ ir::Shape inferTileShape(const ir::Shape &in_shape, const int32_t *multiplier)
   return new_Shape;
 }
 
-ir::Shape inferTransposeShape(const ir::Shape &in_shape, const std::vector<int> &perm)
+ir::Shape inferTransposeShape(const ir::Shape &in_shape, const int32_t *perm, const int32_t rank)
 {
-  if (static_cast<int>(perm.size()) > in_shape.rank())
+  if (rank > in_shape.rank())
```

Comment:
IIRC from NNAPI spec, `rank` can be less than `in_shape.rank()`. But the algorithm seems to not take this into account. Could you please check? A test would be great.

Reply:
The `rank` actually is the number of `perm`. According to NNAPI spec, `perm` is an optional input since V1_2. But our runtime does not support it yet. I will implement it in another PR. And I will rename this in this PR.